overlays/eval-packages.nix

```diff
@@ -0,0 +1,10 @@
+final: prev: {
+  # This overlay makes `evalPackages` is like `buildPackages`, but on
+  # `builtins.currentSystem`.
+  evalPackages =
+    if builtins ? currentSystem
+      then (import final.path {
+          inherit (final.haskell-nix) overlays;
```

Comment:
Just looking at this... this is a little wrong, isn't it? Since we're missing the `config` argument, and also any user-provided overlays. Maybe that's fine, I'm not sure.

Reply:
Yes both config and user provided overlays are left out of evalPackages (when eval is running on the currentSystem).  I am not sure if it is the right thing to do, but that is how [the old code worked too](https://github.com/input-output-hk/haskell.nix/pull/1162/files#diff-5d70876b72bd7f042126980f382e049ab978ae759883bcf9c8a273e706b5b16bL5-L7).

Reply:
That link did not work well, but the old code is in `overlays/eval-on-current.nix`.

Reply:
https://github.com/input-output-hk/haskell.nix/blob/551407d11cd19d5e09e86c876e2919c40e779bb9/overlays/eval-on-current.nix#L5-L7

Reply:
Yeah I know that's how it works currently, I just noticed it there. Maybe add a comment that this is a known thing. And it's a way that eval-on-current and eval-on-build can diverge :shrug: